react-frontend/src/listings/EditListing.js

```diff
@@ -168,6 +172,35 @@ function EditListing() {
             value={listing.title}
             onChange={handleChange}
           />
+          <label htmlFor="category">New Category</label>
+          <select
+          id="category"
+          name="category"
+          onChange={handleChange}>
+            <option value={listing.category}>No Change</option>
+            <option value="Other">Other</option>
+            <option value="Vehicles">Vehicles</option>
+            <option value="Property Rentals">Property Rentals</option>
+            <option value="Apparel">Apparel</option>
+            <option value="Classifieds">Classifieds</option>
+            <option value="Electronics">Electronics</option>
+            <option value="Entertainment">Entertainment</option>
+            <option value="Family">Family</option>
+            <option value="Free Stuff">Free Stuff</option>
+            <option value="Garden & Outdoor">Garden & Outdoor</option>
+            <option value="Hobbies">Hobbies</option>
+            <option value="Home Goods">Home Goods</option>
+            <option value="Home Improvement">Home Improvement</option>
+            <option value="Supplies">Supplies</option>
+            <option value="Home Improvement Supplies">Home Improvement Supplies</option>
+            <option value="Home Sales">Home Sales</option>
+            <option value="Musical Instruments">Musical Instruments</option>
+            <option value="Office Supplies">Office Supplies</option>
+            <option value="Pet Supplies">Pet Supplies</option>
+            <option value="Sporting Goods">Sporting Goods</option>
+            <option value="Toys & Games">Toys & Games</option>
+            <option value="Buy and Sell Groups">Buy and Sell Groups</option>
+          </select>
```

Comment:
**Instead of hard coding the categories as individual tags you should set a categories array:**

`const categories = [
  "No Change",
  "Other",
  "Vehicles",
];`

**Then, where the option tags are change it to use a lambda mapping function:**
`{categories.map((category, index) => (
            <option key={index} value={category}>
              {category}
            </option>
          ))}`

**This will allow for better modularity (only requiring changing the const categories) and cleaner code (less duplicate tags)**

Reply:
Sounds reasonable to me. Due to some strange interactions, the "No Change" block will likely have to remain outside the mapping since it's value is different, but the rest should integrate just fine.

Reply:
I think this would be a valuable change to make and supports scaling

Reply:
+1

Reply:
supplies , office supplies, and home improvement supplies may be a bit overkill and can all be combined.